src/interpreter.js

```diff
@@ -0,0 +1,61 @@
+const bscript = require('bscript-parser')
+
+const Interpreter = module.exports = class Interpreter {
+  constructor (script) {
+    this.script = script
+    this.pc = 0
+    this.pend = script.length
+    this.stack = []
+  }
+
+  step () {
+    if (this.pc >= this.pend) {
+      throw Error('Cannot step, no further instructions')
+    }
+    if (typeof this.script === 'undefined' || this.script.length === 0) {
+      throw Error('No script provided')
+    }
+    // Validate opcode
+    if (!bscript.opcodes.opcodeIsValid(this.script[this.pc])) {
+      throw Error(`Invalid opcode ${this.script[this.pc].toString(16)}`)
+    } else {
+      if (this.script[this.pc] <= 0x4b) {
+        this.stack.push(this.script[this.pc])
```

Comment:
Is this correct?  I may not understand what this is doing.  It looks to me like you are just pushing the number of bytes to push rather than the actual bytes that are supposed to be pushed.  Am I misunderstanding?

Reply:
Oh yeah nice catch, it should be pushing that many bytes, not the number itself.